cmd/tools/rest/HrefBuilder.go

```diff
@@ -0,0 +1,86 @@
+package rest
+
+import (
+	"strconv"
+	"strings"
+)
+
+type HrefBuilder struct {
+	apiPath       string
+	fields        string
+	filter        []string
+	queryFields   string
+	queryValue    string
+	maxRecords    int
+	returnTimeout *int
+}
+
+func NewHrefBuilder() *HrefBuilder {
+	return &HrefBuilder{}
+}
+
+func (b *HrefBuilder) APIPath(apiPath string) *HrefBuilder {
+	b.apiPath = apiPath
+	return b
+}
+
+func (b *HrefBuilder) Fields(fields []string) *HrefBuilder {
+	b.fields = strings.Join(fields, ",")
+	return b
+}
+
+func (b *HrefBuilder) Filter(filter []string) *HrefBuilder {
+	b.filter = filter
+	return b
+}
+
+func (b *HrefBuilder) QueryFields(queryFields string) *HrefBuilder {
+	b.queryFields = queryFields
+	return b
+}
+
+func (b *HrefBuilder) QueryValue(queryValue string) *HrefBuilder {
+	b.queryValue = queryValue
+	return b
+}
+
+func (b *HrefBuilder) MaxRecords(maxRecords int) *HrefBuilder {
+	b.maxRecords = maxRecords
+	return b
+}
+
+func (b *HrefBuilder) ReturnTimeout(returnTimeout *int) *HrefBuilder {
```

Comment:
It's a good idea that our API understands whether a return timeout was set or not, but we shouldn't require consumers of this API to pass a pointer, passing an int is friendly